src/services/db/GitFileSystemService.ts

```diff
@@ -369,13 +370,14 @@ export default class GitFileSystemService {
   // Read the contents of a file
   read(
     repoName: string,
-    filePath: string
+    filePath: string,
+    encoding: "utf-8" | "base64" = "utf-8"
```

Comment:
nit: Hmm is it a lot of effort to remove the default? would rather enforce callers to define encoding for clarity!

Reply:
i see this as by default you should be always reading as utf-8 for all cases, except the case for images when base64 is needed. Hence I made it optional

If we find ourselves adding more encoding types in future we can make this explicit, but for now base64 seems to be the special case